dockerfiles/cert-manager-ca-cert-generator-job/entrypoint.sh

```diff
@@ -0,0 +1,36 @@
+#!/bin/sh
+
+CA_KEY_FILE='ca.key'
+CA_CERT_FILE='ca.crt'
+
+# Generate private key for root CA
+# Options:
+#  -out : name of file to write generated key to
+#  4096 : number of bits in the key
+openssl genrsa -out $CA_KEY_FILE 4096
+
+# Generate CA certificate and sign it with previously generated key.
+# Options:
+#  -batch : script (non-interactive) mode
+#  -new : creates new sertificate request
+#  -x509 : produces self signed sertificate instead of certificate request
+#  -deys : number of days this certificate will be valid for
+#  -key : private key to use to sign this certificate
+#  -subj : subject name. Should contain at least distinguished (common) name (CN). Format: /type0=value0/type1=value1
+#  -addext : adds extension to certificate (inline version of -reqexts with -config)
+#  -outform : format of the certificate container
+#  -out : name of file to write generated certificate to
+CA_CN='eclipse-che-local-CA'
+openssl req -batch -new -x509 -days 730 -key $CA_KEY_FILE \
+            -subj "/CN=${CA_CN}" \
+            -addext keyUsage=keyCertSign,cRLSign,digitalSignature \
+            -outform PEM -out $CA_CERT_FILE
+            # Do not include CA:TRUE as it is already included into default config file
+            #-addext basicConstraints=critical,CA:TRUE
+
+# Create CA root certificate secret
+
+CERT_MANAGER_NAMESPACE='cert-manager'
+CERT_MANAGER_CA_SECRET_NAME='ca'
+
+kubectl create secret tls $CERT_MANAGER_CA_SECRET_NAME --key=$CA_KEY_FILE --cert=$CA_CERT_FILE --namespace $CERT_MANAGER_NAMESPACE
```

Comment:
So.. .this container is just kubectl + a cert? Instead of new container, could we instead just add this cert into https://github.com/che-dockerfiles/che-sidecar-kubernetes-tooling and keep the proliferation of containers (and therefore things that need to be productized) to a minimum? CRW 1.2 was 8 containers. 2.0 was 20. 2.1 is up to 23 and counting. Can we try to reuse out existing tiny containers and just add a new kb of new files into them??

Reply:
@nickboldt yes, this container indeed contains kubectl and openssl only, but we cannot reuse Kubernetes tolling sidecar for the Cert Manager job. First, they have different purposes: the job completes its tasks and exists, whereas Kubernetes tooling sidecar is designed to be run as a part of a workspace. Second, they have different aims: generate certificate and serve queries to Kubernetes respectively. Third, chectl (and Che installation process) should not depend on a Che sidecar. That's logically wrong and error prone: if something is changed in Kubernetes Che plugin it could break Che installation.

Reply:
*sigh* OK, I was hoping for reuse and less proliferation of single-use containers. Your logic is sound, but still makes me sad. :(